### PR TITLE
#18 - Asset, Config  and PagePredicate models now leverage the modelC…

### PR DESCRIPTION
**Pull request by davidjgonzalez**
…ache so they are not reinitialized for every component on the page that uses them.

@godanny86 would be good to have an extra sanity check that the sample content works and i didn't miss testing any of the components.

### CODE REVIEW

File: ui.apps/src/main/content/jcr_root/apps/asset-share-commons/components/details/action-buttons/action-buttons.html [hunk 1]
```diff
@@ -16,9 +16,10 @@
   ~ limitations under the License.
  */-->
 
-<sly data-sly-use.config="com.adobe.aem.commons.assetshare.configuration.Config"
+<sly data-sly-use.modelCache="com.adobe.aem.commons.assetshare.util.ModelCache"
+     data-sly-test.config="${modelCache['com.adobe.aem.commons.assetshare.configuration.Config']}"
+     data-sly-test.asset="${modelCache['com.adobe.aem.commons.assetshare.content.AssetModel']"
```
Comment (godanny86) [inline ui.apps/src/main/content/jcr_root/apps/asset-share-commons/components/details/action-buttons/action-buttons.html]: missing closing '}'

File: ui.apps/src/main/content/jcr_root/apps/asset-share-commons/components/search/results/result/card/card.html [hunk 2]
```diff
@@ -17,9 +17,9 @@
  */-->
 
 <sly data-sly-use.modelCache="com.adobe.aem.commons.assetshare.util.ModelCache"
-     data-sly-use.pagePredicate="com.adobe.aem.commons.assetshare.components.predicates.PagePredicate"
      data-sly-test.config="${modelCache['com.adobe.aem.commons.assetshare.configuration.Config']}"
      data-sly-test.search="${modelCache['com.adobe.aem.commons.assetshare.search.Search']}"
+     data-sly-test.pagePredicate="${modelCache['com.adobe.aem.commons.assetshare.components.predicates.PagePredicate']"
```
Comment (godanny86) [inline ui.apps/src/main/content/jcr_root/apps/asset-share-commons/components/search/results/result/card/card.html]: missing closing '}'

File: ui.apps/src/main/content/jcr_root/apps/asset-share-commons/components/search/results/result/list/list.html [hunk 3]
```diff
@@ -17,9 +17,9 @@
  */-->
 
 <sly data-sly-use.modelCache="com.adobe.aem.commons.assetshare.util.ModelCache"
-	 data-sly-use.pagePredicate="com.adobe.aem.commons.assetshare.components.predicates.PagePredicate"
 	 data-sly-test.config="${modelCache['com.adobe.aem.commons.assetshare.configuration.Config']}"
 	 data-sly-test.search="${modelCache['com.adobe.aem.commons.assetshare.search.Search']}"
+	 data-sly-test.pagePredicate="${modelCache['com.adobe.aem.commons.assetshare.components.predicates.PagePredicate']"
```
Comment (godanny86) [inline ui.apps/src/main/content/jcr_root/apps/asset-share-commons/components/search/results/result/list/list.html]: missing closing '}'

File: ui.apps/src/main/content/jcr_root/apps/asset-share-commons/components/search/results/results.html [hunk 4]
```diff
@@ -16,10 +16,10 @@
...
  */-->
 
-<sly    data-sly-use.config="com.adobe.aem.commons.assetshare.configuration.Config"
-        data-sly-use.predicate="com.adobe.aem.commons.assetshare.components.predicates.PagePredicate"
-        data-sly-use.modelCache="com.adobe.aem.commons.assetshare.util.ModelCache"
-        data-sly-test.search="${modelCache['com.adobe.aem.commons.assetshare.search.Search']}"></sly>
+<sly    data-sly-use.modelCache="com.adobe.aem.commons.assetshare.util.ModelCache"
+        data-sly-test.config="${modelCache['com.adobe.aem.commons.assetshare.configuration.Config']}"
+        data-sly-test.search="${modelCache['com.adobe.aem.commons.assetshare.search.Search']}"
+        data-sly-test.pagePredicate="${modelCache['com.adobe.aem.commons.assetshare.components.predicates.PagePredicate']"></sly>
```
Comment (godanny86) [inline ui.apps/src/main/content/jcr_root/apps/asset-share-commons/components/search/results/results.html]: missing closing '}'

**Comment by davidjgonzalez** [comment]
@godanny86 should be fixed

**Pull request merged**